### ISSUE TITLE
ConvolutionMM Module is not getting train with Image Data

### ISSUE THREAD

**Issue by jbansal** [torch/cutorch#231]
Hi ,
I am using ConvolutionMM to train the machine on Image data for my project.
Model is defined as below:

require 'cudnn'
require 'cunn'
model=nn.Sequential()

model:add(cudnn.SpatialConvolution(3,32, 2, 2))
model:add(cudnn.Tanh())
model:add(cudnn.SpatialMaxPooling(2, 2,2,2))

-- stage 2 : filter bank -> squashing -> max pooling
model:add(cudnn.SpatialConvolution(32, 64, 2, 2))
model:add(cudnn.Tanh())
model:add(cudnn.SpatialMaxPooling(2, 2))
 -- stage 3 : standard 2-layer neural network
model:add(nn.Reshape(64*25*25))
model:add(nn.Tanh())
model:add(nn.Linear(64*25*25,10))-- classify in 10 category
model:add(nn.Tanh())
model:add(nn.LogSoftMax())

local epsilonLayer1 = math.sqrt(6)/(math.sqrt(105*105+32*32) * math.sqrt(3))
local epsilonLayer2 = math.sqrt(6)/(math.sqrt(32*32+64*64) * math.sqrt(3))
local epsilonLayer3 = math.sqrt(6)/(math.sqrt(64*64+10) * math.sqrt(3))
model:get(1):reset(epsilonLayer1)
model:get(4):reset(epsilonLayer2)
model:get(9):reset(epsilonLayer3)

I am using 	criterion = nn.ClassNLLCriterion()

When I checked my loss it is not improving and values of gradBias  and gradWeights are calculating very high e+40 or e-40 and some times as nan/inf.

I am using model as follows:
here inputs are cuda tensor of n*3*width*height. where n is number of image inputs are passed in the 
local guess =model:forward(inputs,targets)
local loss_x  =  criterion:forward(guess, targets)
loss =loss+loss_x
local gradInput = criterion:backward(guess,targets)
model:backward(inputs,gradInput)

loss of this criterion is coming as follows:
currentepoch=1,loss =0.009278, time =31.345217	
currentepoch=2,loss =0.010519, time =31.907952	
currentepoch=3,loss =0.011127, time =31.518895	
currentepoch=4,loss =0.010324, time =29.424960	
currentepoch=5,loss =0.011531, time =29.422763	
currentepoch=6,loss =0.011513, time =29.037167	
currentepoch=7,loss =0.011512, time =29.903065	
currentepoch=8,loss =0.011512, time =30.261979	
currentepoch=9,loss =0.011512, time =30.391334	

after some epochs, loss is not decreasing or loss is moving in some range.

I am not able to find why this issue is happening. Is it due to some internal issue in module or criterion or something wrong with my data or logic.

Any help/clue highly appreciated.

Regards,
Jitendra Bansal

**Comment by fmassa** [comment]
Those questions are best for the google groups https://groups.google.com/forum/#!forum/torch7

For your case, I'd say you should probably remove the last `Tanh` just before `LogSoftMax`.

If it isn't enough, open an issue in the google groups and I'll help you there.
